sdk/ml/azure-ai-ml/azure/ai/ml/operations/_capability_hosts_operations.py

```diff
@@ -0,0 +1,345 @@
+        """
+
+        self.__validate_workspace_name()
+
+        rest_obj = self._capability_hosts_operations.get(
+            resource_group_name=self._resource_group_name,
+            workspace_name=self._workspace_name,
+            name=name,
+            **kwargs,
+        )
+
+        capability_host = CapabilityHost._from_rest_object(rest_obj)
+
+        return capability_host
+
+    @experimental
+    @monitor_with_activity(ops_logger, "CapabilityHost.BeginCreateOrUpdate", ActivityType.PUBLICAPI)
+    @distributed_trace
+    def begin_create_or_update(self, capability_host: CapabilityHost, **kwargs: Any) -> LROPoller[CapabilityHost]:
+        """Begin the creation of a capability host in a Hub or Project workspace.
+        Note that currently this method can only accept the `create` operation request
+        and not `update` operation request.
+
+        :param capability_host: The CapabilityHost object containing the details of the capability host to create.
+        :type capability_host: ~azure.ai.ml.entities.CapabilityHost
+        :return: An LROPoller object that can be used to track the long-running
+            operation that is creation of capability host.
+        :rtype: ~azure.core.polling.LROPoller[~azure.ai.ml.entities._workspace._ai_workspaces.capability_host.CapabilityHost]    # pylint: disable=line-too-long
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/ml_samples_capability_host.py
+                :start-after: [START capability_host_begin_create_or_update_operation]
+                :end-before: [END capability_host_begin_create_or_update_operation]
+                :language: python
+                :dedent: 8
+                :caption: Create example.
+        """
+        try:
+            self.__validate_workspace_name()
+
+            workspace = self.__get_workspace()
+
+            self.__validate_properties(capability_host, workspace._kind)
+
+            if workspace._kind == WorkspaceKind.PROJECT:
+                if capability_host.storage_connections is None or len(capability_host.storage_connections) == 0:
+                    capability_host.storage_connections = self.__get_default_storage_connections()
+
+            capability_host_resource = (
+                capability_host._to_rest_object_for_hub()
+                if workspace._kind == WorkspaceKind.HUB
+                else capability_host._to_rest_object_for_project()
+            )
+
+            # pylint: disable=unused-argument, docstring-missing-param
+            def callback(_: Any, deserialized: Any, args: Any) -> CapabilityHost:
+                """Callback to be called after completion of begin_create_or_update operation
+
+                :return: Deserialized CapabilityHost.
+                :rtype: ~azure.ai.ml.entities._workspace._ai_workspaces.capability_host.CapabilityHost
+                """
+
+                return CapabilityHost._from_rest_object(deserialized)
```

Comment:
use the `lambda` from the suggestion I gave above instead of this `callback`

Reply:
done